tools/py/gleacf.py

```diff
@@ -145,61 +145,83 @@ def ISRA(omega, ker, y, dparam, oprefix):
     return f
 
 
-def gleacf(path2iipi, path2ivvac, oprefix, action, nrows, stride, dparam):
-
-    # opens & parses the i-pi input file
-    ifile = open(path2iipi, "r")
-    xmlrestart = io_xml.xml_parse_file(ifile)
-    ifile.close()
-
-    isimul = InputSimulation()
-    isimul.parse(xmlrestart.fields[0][1])
-    simul = isimul.fetch()
-
-    # parses the drift and diffusion matrices of the GLE thermostat.
-    ttype = str(type(simul.syslist[0].motion.thermostat).__name__)
-    kbT = float(simul.syslist[0].ensemble.temp)
-    simul.syslist[0].motion.thermostat.temp = kbT
-
-    if(ttype == "ThermoGLE"):
-        Ap = simul.syslist[0].motion.thermostat.A * unit_to_internal("time", dt[1], float(dt[0]))
-        Cp = simul.syslist[0].motion.thermostat.C / kbT
+def gleacf(path2ixml, path2iA, path2iC, path2ifacf, oprefix, action, nrows, stride, tscale, dparam):
+
+    if path2ixml != None and path2iA != None:
+        raise Exception("The drift and covariance matrices have to be provided either through the i-pi xml file or manually. Can not use both forms of input simultaneously. ")
+
+    elif path2ixml != None and path2iA == None:
+        # opens & parses the i-pi input file
+        ifile = open(path2ixml, "r")
+        xmlrestart = io_xml.xml_parse_file(ifile)
+        ifile.close()
+
+        isimul = InputSimulation()
+        isimul.parse(xmlrestart.fields[0][1])
+        simul = isimul.fetch()
+
+        # parses the drift and diffusion matrices of the GLE thermostat.
+        ttype = str(type(simul.syslist[0].motion.thermostat).__name__)
+        P =  float(simul.syslist[0].init.nbeads)
+        kbT = float(simul.syslist[0].ensemble.temp) * P
+        simul.syslist[0].motion.thermostat.temp = kbT
+
+        if(ttype == "ThermoGLE"):
+            Ap = simul.syslist[0].motion.thermostat.A * tscale
+            Cp = simul.syslist[0].motion.thermostat.C / kbT
+            Dp = np.dot(Ap, Cp) + np.dot(Cp, Ap.T)
+        if(ttype == "ThermoNMGLE"):
+            Ap = simul.syslist[0].motion.thermostat.A[0] * tscale
+            Cp = simul.syslist[0].motion.thermostat.C[0] / kbT
+            Dp = np.dot(Ap, Cp) + np.dot(Cp, Ap.T)
+        elif(ttype == "ThermoLangevin"):
+            Ap = np.asarray([1.0 / simul.syslist[0].motion.thermostat.tau]).reshape((1, 1)) * tscale 
+            Cp = np.asarray([1.0]).reshape((1, 1))
+            Dp = np.dot(Ap, Cp) + np.dot(Cp, Ap.T)
```

Comment:
Add PILE-L, and an else: raise ValueError of some kind

Reply:
Added.